cc2dataset/main.py

```diff
@@ -53,58 +56,73 @@ def extract_video_from_links(links):
 )
 
 
-def valid_text_link(link):
-    if not link.get("url", "").startswith("http"):
-        return False
+def valid_text_link(link, base_url):
     splits = link.get("url", "").split(".")
     if len(splits) < 2:
         return False
     if splits[-1] not in text_extensions:
         return False
-    return True
+    link_make_absolute_url(link, base_url, "url")
+    return link.get("url", "").startswith("http")
 
 
-def extract_text_from_links(links):
-    filtered_links = [{"url": link["url"], "alt": link.get("text", "")} for link in links if valid_text_link(link)]
+def extract_text_from_links(links, base_url):
+    filtered_links = [{"url": link["url"], "alt": link.get("text", "")} for link in links if valid_text_link(link, base_url)]
     return filtered_links
 
 
-def valid_audio_link(link):
-    valid_http = link.get("url", "").startswith("http")
+def valid_audio_link(link, base_url):
     valid_audio = any(link.get("url", "").endswith(ext) for ext in [".ogg", ".wav", ".mp3", ".flac", ".m4a"])
-    return valid_http and valid_audio
+    if valid_audio:
+        return False
+    link_make_absolute_url(link, base_url, "url")
+    return link.get("url", "").startswith("http")
 
 
-def extract_audio_from_links(links):
+def extract_audio_from_links(links, base_url):
     """Extract image from links"""
-    filtered_links = [{"url": link["url"], "alt": link.get("text", "")} for link in links if valid_audio_link(link)]
+    filtered_links = [{"url": link["url"], "alt": link.get("text", "")} for link in links if valid_audio_link(link, base_url)]
     return filtered_links
 
 
-def valid_image_link(link):
+def valid_image_link(link, base_url):
     valid_path = link.get("path", "") == "IMG@/src"
     valid_alt = len(link.get("alt", "")) > 0
-    valid_http = link.get("url", "").startswith("http")
-    return valid_path and valid_http and valid_alt
+    if not (valid_path and valid_alt):
+        return False
+    link_make_absolute_url(link, base_url, "url")
+    return link.get("url", "").startswith("http")
 
 
-def extract_image_from_links(links):
+def extract_image_from_links(links, base_url):
     """Extract image from links"""
-    filtered_links = [{"url": link["url"], "alt": link["alt"]} for link in links if valid_image_link(link)]
+    filtered_links = [{"url": link["url"], "alt": link["alt"]} for link in links if valid_image_link(link, base_url)]
```

Comment:
here we are keeping the relative links but not making them absolute links. This is going to make further steps fail on these relative links.
I think we should rather make them absolute

Reply:
ah nevermind, I see link_make_absolute_url is changing the link dictionary